Source/WebCore/Modules/model-element/HTMLModelElement.cpp

```diff
@@ -1172,8 +1171,10 @@ Node::InsertedIntoAncestorResult HTMLModelElement::insertedIntoAncestor(Insertio
 {
     auto insertResult = HTMLElement::insertedIntoAncestor(insertionType, parentOfInsertedTree);
 
-    if (insertionType.connectedToDocument)
+    if (insertionType.connectedToDocument) {
+        document().registerForVisibilityStateChangedCallbacks(*this);
```

Comment:
Now you can remove the calls to `registerForVisibilityStateChangedCallbacks` and `unregisterForVisibilityStateChangedCallbacks` in `HTMLModelElement::didMoveToNewDocument`.